gwdetchar/io/datafind.py

```diff
@@ -178,23 +179,22 @@ def get_data(channel, start, end, obs=None, frametype=None, source=None,
         series_class = TimeSeriesDict
     else:
         series_class = TimeSeries
-    # construct file cache if none is given
-    if source is None:
-        obs = obs if obs is not None else frametype[0]
-        source = gwdatafind.find_urls(obs, frametype, start, end)
-    # read from frames or NDS
-    if source:
+    try:  # read from frame files
+        source = source or gwdatafind.find_urls(
+            obs or frametype[0], frametype, start, end)
         if isinstance(channel, (list, tuple)):
             channel = remove_missing_channels(channel, source)
         return series_class.read(
             source, channel, start=start, end=end, nproc=nproc,
             verbose=verbose, **kwargs)
-    elif isinstance(channel, str):
-        return series_class.fetch(
+    except (HTTPError, TypeError):  # frame files not found
```

Comment:
Can you explain when an `HTTPError` would be raised?

Reply:
@duncanmmacleod, the `HTTPError` is raised whenever `gwdatafind.find_urls` can't locate frame files (you get a `400 Bad Request`).

Reply:
I think that if you don't pass `ops` and the first letter of `frametype` is not a meaningful observatory code (e.g. for SenseMon frames), this will also lead to `HTTPError`. The `TypeError` case comes when you pass neither `frametype` nor `source`. In all these cases, the idea is that it would then fall back to `TimeSeries.get` with no protection against missing channels in frame files.

Reply:
@duncanmmacleod, I stand corrected, if you attempt to read SenseMon frames with `frametype='SenseMonitor_hoft_L1_M'` and `obs=None` then the result is an `IORegistryError: Format could not be identified.` Passing `frametype=None` falls back to `TimeSeries.get`, and passing both `frametype` and `obs` successfully uses `TimeSeries.read`.